lib/skate/settings/test_group.ex

```diff
@@ -12,12 +12,20 @@ defmodule Skate.Settings.TestGroup do
 
   defstruct [:id, :name, users: []]
 
-  @spec create(String.t()) :: __MODULE__.t()
+  @spec create(String.t()) :: {:ok, t()} | {:error, Ecto.Changeset.t()}
   def create(name) do
-    %DbTestGroup{name: name}
-    |> Skate.Repo.insert!()
-    |> Skate.Repo.preload(:users)
-    |> convert_from_db_test_group()
+    case %DbTestGroup{}
+         |> DbTestGroup.changeset(%{name: name})
+         |> Skate.Repo.insert() do
```

Comment:
suggestion(non-blocking): AFAIK, you can add the case as another pipeline entry, and I think it's easier to read this way.
```suggestion
    %DbTestGroup{}
    |> DbTestGroup.changeset(%{name: name})
    |> Skate.Repo.insert() 
    |> case do
```

Reply:
Oh cool - I didn't know you could do that